### PR TITLE
Better search relevancy: symmetrical to both query and target text.

### PR DESCRIPTION
**Pull request by isoos**
Started as a follow-up to yesterdays review, but found an easy way to solve many search quirks at once (line 186 was the key, the others are only fine-tuning). Also added tests for the edge cases that document the underlying scores.

### CODE REVIEW

File: app/lib/search/index_simple.dart [hunk 1]
```diff
@@ -62,64 +62,70 @@ class SimplePackageIndex implements PackageIndex {
 
   @override
   Future<PackageSearchResult> search(PackageQuery query) async {
-    final Map<String, _PackageResult> resultMap = {};
+    final Map<String, double> total = {};
```
Comment (kevmoo) [inline app/lib/search/index_simple.dart]: For future – put types on the right for literals.

File: app/lib/search/index_simple.dart [hunk 2]
```diff
@@ -62,64 +62,70 @@ class SimplePackageIndex implements PackageIndex {
...
-            !doc.detectedTypes.contains(query.type);
-      });
-    }
+    final Map<String, double> popularityScores = new Map.fromIterable(
+      total.keys,
+      value: (String url) => _documents[url].popularity * 100,
+    );
+    addAll(popularityScores, 0.05);
+
+    List<PackageScore> results = [];
```
Comment (kevmoo) [inline app/lib/search/index_simple.dart]: ditto – var results = <PackageScore>[];

**Comment by isoos** [comment]
> For future – put types on the right for literals.

Fixed the ones in this file. Shall we turn on `prefer_collection_literals`?

**Comment by kevmoo** [comment]
> Fixed the ones in this file. Shall we turn on prefer_collection_literals?

Sure!
